task/runner.go

```diff
@@ -545,6 +546,97 @@ func (r *runner) Shutdown(ctx context.Context) error {
 	return r.amqp.Shutdown(ctx)
 }
 
+func (r *runner) UnpinFromIpfs(ctx context.Context, cid string, filter string) error {
+	assets, _, err := r.lapi.ListAssets(api.ListOptions{
+		Filters: map[string]interface{}{
+			filter: cid,
+		},
+		AllUsers: true,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if len(assets) == 1 {
+		return r.ipfs.Unpin(ctx, cid)
+	}
+
+	return nil
+}
+
+func (r *runner) CronJobForAssetDeletion(ctx context.Context) error {
+	// Loop every hour to delete assets
+	ticker := time.NewTicker(60 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			assets, err := r.lapi.GetDeletingAssets()
+			if err != nil {
+				glog.Errorf("Error retrieving assets for deletion: %v", err)
+				continue
+			}
+			for _, asset := range assets {
+				_, _, assetOS, err := r.getAssetAndOS(asset.ID)
+				if err != nil {
+					glog.Errorf("Error getting asset object store session: %v", err)
+					continue
+				}
+
+				directory := asset.PlaybackID
+				var totalDeleted int
+
+				// Initially list files
+				pi, err := assetOS.ListFiles(ctx, directory, "/")
+				if err != nil {
+					glog.Errorf("Error listing files for asset %v: %v", asset.ID, err)
+					continue
+				}
+
+				for pi != nil {
+					for _, file := range pi.Files() {
+						err := assetOS.DeleteFile(ctx, file.Name)
+						if err != nil {
+							glog.Errorf("Error deleting file %v: %v", file.Name, err)
```

Comment:
Reminder to not tag the asset as deleted if there are any errors deleting file

Reply:
As a general technique I like when creating background jobs like this, you could create an internal version of the function that returns an error like other functions normally do, then have a "background runner" function that only handles calling that function and logging whatever error it returns. It generally makes the code less repetitive with all the logs and less error-prone (e.g. if you do an early return on sth that should log). 

No need to implement this tho, sending as a suggestion in case you find it useful when you refactor it to a separate file/out of `runner`